staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -233,21 +247,22 @@ func (d *DiscoveryClient) downloadLegacy() (*metav1.APIGroupList, map[schema.Gro
 	if err != nil {
 		// Tolerate 404, since aggregated api servers can return it.
 		if errors.IsNotFound(err) {
-			return &metav1.APIGroupList{}, nil, nil
+			return &metav1.APIGroupList{}, nil, nil, nil
```

Comment:
if `err == nil` shouldn't the `failedGVs` be non-nil, so you can add to it a few lines up in the diff?  If so, I think this would indicate a missing unit test.

Reply:
You are correct...and digging into this I found that aggregated discovery was not correctly tolerating `404` for `core/v1`. I have rectified this, and added a new unit test. Please have a look.